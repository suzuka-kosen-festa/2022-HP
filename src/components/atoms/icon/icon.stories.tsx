import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import tw from "twin.macro";
import Icon from ".";

type T = typeof Icon;
type Story = ComponentStoryObj<T>;

const data = {
  icon: "ant-design:warning-filled",
  iconStyles: tw`text-caution`,
};

export default {
  args: { icon: data.icon, iconStyles: data.iconStyles },
  argTypes: {
    icon: {
      description: "Icon key for iconify",
      control: {
        type: "text",
      },
    },
    iconStyles: {
      description: "Icon style",
      control: {
        type: "object",
      },
    },
    as: {
      description: "As properties for component",
      control: {
        type: "text"
      }
    },
    css: {
      description: "Icon box style",
      control: {
        type: "object",
      },
    },
  },
  component: Icon,
} as ComponentMeta<T>;

export const Default: Story = {};
