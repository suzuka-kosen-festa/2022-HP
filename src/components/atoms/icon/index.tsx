import { InlineIcon } from "@iconify/react";
import { memo } from "react";
import tw from "twin.macro";
import type { IconProperties } from "./type/model";

const IconBox = tw.div`flex items-center justify-center flex-shrink-0 w-6 h-6 sm:(w-12 h-12) min-h-full`;

const Icon = <T extends AnyComponent>({ icon, iconStyles, as, ...rest }: IconProperties<T>): JSX.Element => (
  <IconBox as={as} {...rest}>
    <InlineIcon icon={icon} css={[tw`w-full h-full`, iconStyles]} />
  </IconBox>
);

export default memo(Icon);
