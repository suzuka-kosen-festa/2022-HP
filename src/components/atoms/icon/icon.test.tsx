import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./icon.stories";

const { Default } = composeStories(stories);

describe("(components) atoms/icon", () => {
  test("to be atoms", () => {
    const { container } = render(<Default />);
    expect(container).toBeAtom();
  });
  test("take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
