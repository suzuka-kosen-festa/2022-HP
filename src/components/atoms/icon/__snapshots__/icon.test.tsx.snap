// Vitest Snapshot v1

exports[`(components) atoms/icon > take snap shot 1`] = `
<div>
  <div
    class="css-1vlcokg"
  >
    <span />
  </div>
</div>
`;
